aten/src/ATen/native/Math.h

```diff
@@ -1385,37 +1464,6 @@ calc_i0(T _x) {
 // Upcast bfloat16 input to float for numerical accuracy purposes
 static inline c10::BFloat16 calc_i0(c10::BFloat16 a) { return calc_i0(static_cast<float>(a)); }
 
-/*
- * This function is derived from the implementation of the i0e function in the Cephes Math Library.
- * See note [3-Clause BSD License for the Cephes Math Library].
- *
- * Computes an approximation of the exponentially scaled zeroth order modified Bessel function of the first kind.
- * The approximation is actually two (sub)approximations, both using a Chebyshev polynomial expansion.
- * One approximates the function over [0, 8], and the other over (8, infinity). This function takes the absolute value
- * of all inputs to convert them into the domain of the approximation.
- */
-template <typename T>
-static inline typename std::enable_if<std::is_floating_point<T>::value, T>::type
-calc_i0e(T _x) {
-  T x = std::abs(_x);
-
-  if (x <= T{8.0}) {
-    auto coeff_pair = chebyshev_coefficients_i0e_A<T>();
-    auto A = std::get<0>(coeff_pair);
-    auto len = std::get<1>(coeff_pair);
-    T y = (x / T{2.0}) - T{2.0};
-    return chbevl(y, A, len);
-  }
-
-  auto coeff_pair = chebyshev_coefficients_i0e_B<T>();
-  auto B = std::get<0>(coeff_pair);
-  auto len = std::get<1>(coeff_pair);
-  return chbevl(T{32.0} / x - T{2.0}, B, len) / std::sqrt(x);
-}
-
-// Upcast bfloat16 input to float for numerical accuracy purposes
-static inline c10::BFloat16 calc_i0e(c10::BFloat16 a) { return calc_i0e(static_cast<float>(a)); }
```

Comment:
Where did all this code go?

Reply:
Jiterated version of the code has the coefficients (which we get from `chebyshev_coefficients_i0e_B` and `chebyshev_coefficients_i0e_A`)
And BFloat16 upcasting here was redundant as it is handled at other places.

Existing tests in test_unary_ufuncs verify the correctness against scipy implementation for CPU and CUDA.